lib/types/SSITypesBuilder.ts

```diff
@@ -46,14 +46,56 @@ export class SSITypesBuilder {
   private static mapExternalVerifiableCredentialToInternal(externalCredential: IVerifiableCredential) {
     if (externalCredential.vc && externalCredential.iss) {
       const vc: InternalVerifiableCredential = new InternalVerifiableCredentialJwt();
-      return Object.assign(vc, externalCredential);
+      const result = Object.assign(vc, externalCredential);
+      return SSITypesBuilder.setJWTAdditionalFields(result);
     } else if (externalCredential.credentialSubject && externalCredential.id) {
       const vc: InternalVerifiableCredential = new InternalVerifiableCredentialJsonLD();
       return Object.assign(vc, externalCredential);
     }
     throw 'VerifiableCredential structure is incorrect.';
   }
 
+  private static setJWTAdditionalFields(result: InternalVerifiableCredentialJwt & IJwtCredential & IHasProof) {
+    if (result.exp) {
+      const expDate = result.getBaseCredential().credentialSubject?.expirationDate;
+      if (expDate && expDate !== result.exp) {
+        throw new Error(`Inconsistent expiration dates between JWT claim (${result.exp} and VC value (${expDate})`);
+      }
+      const exp = result.exp.match(/^\d+$/) ? parseInt(result.exp) : result.exp;
+      result.getBaseCredential().credentialSubject.expirationDate = new Date(exp).toISOString();
+    }
+    if (result.iss) {
+      const issuer = result.getBaseCredential().issuer;
+      if (issuer && issuer !== result.iss) {
+        throw new Error(`Inconsistent issuers between JWT claim (${result.iss} and VC value (${issuer})`);
```

Comment:
consider using PEMessages instead of literal strings in this file (and tests)

Reply:
Good one, forgot about that change

Reply:
Cannot use PEMessages because the strings contain computed values.

Reply:
You could use someting like util.format. But that probably brings problems for RN environments, where a polyfill would be needed. So lets keep it simple for now